libarchive/archive_read_support_format_zip.c

```diff
@@ -899,6 +899,59 @@ process_extra(struct archive_read *a, struct archive_entry *entry,
 	return ARCHIVE_OK;
 }
 
+/*
+ * Auxiliary method to uncompress data chunk from zipx archive (zip with lzma compression).
+ */
+static int
+zipx_lzma_uncompress_buffer(const char *compressed_buffer, size_t compressed_buffer_size,
+  char *uncompressed_buffer, size_t uncompressed_buffer_size)
+{
+  int status = ARCHIVE_FATAL;
+  if (compressed_buffer == NULL || compressed_buffer_size < 9 || uncompressed_buffer == NULL)
+    return status;
+
+  // prepare header for lzma_alone_decoder to replace zipx header (see comments in 'zipx_lzma_alone_init' for justification)
+#pragma pack(push)
+#pragma pack(1)
+  struct _alone_header
+  {
+    uint8_t bytes[5];
+    uint64_t uncompressed_size;
+  } alone_header;
+#pragma pack(pop)
+
+  memcpy(&alone_header.bytes[0], compressed_buffer + 4, 5);
+  alone_header.uncompressed_size = UINT64_MAX;
+
+  // prepare new compressed buffer, see 'zipx_lzma_alone_init' for details
+  int lzma_alone_buffer_size = compressed_buffer_size - 9 + sizeof(alone_header);
```

Comment:
I'd change the `int` to `size_t` to get rid of mismatched type warnings (since `compressed_buffer_size` is a `size_t`, as well as `avail_in` is also `size_t`)

Reply:
done